spark/src/main/scala/org/apache/spark/sql/comet/CometBatchScanExec.scala

```diff
@@ -144,11 +144,5 @@ case class CometBatchScanExec(wrapped: BatchScanExec, runtimeFilters: Seq[Expres
     }
   }
 
-  // Intentionally omitting the return type as it is different depending on Spark version
-  // Spark 3.2.x Seq[InputPartition]
-  // Spark 3.3.x Seq[Seq[InputPartition]]
-  // TODO: add back the return type after dropping Spark 3.2.0 support
-  @transient override lazy val partitions = wrappedScan.partitions
```

Comment:
Based on the comment I would expect this to still be needed but can now have a return type. But maybe it no longer needed?
```
 @transient override lazy val partitions: Seq[Seq[InputPartition]] = wrappedScan.partitions
 ```

Reply:
@eejbyfeldt This should still be needed. I accidentally removed this.